components/mediation-admin/org.wso2.carbon.rest.api/src/main/java/org/wso2/carbon/rest/api/service/RestApiAdmin.java

```diff
@@ -928,5 +958,173 @@ protected ConfigurationContext getConfigContext() {
+     * @param tenantId
+     * @throws APIException
+     */
+    public void updateSwaggerDocument(String apiName, String swaggerJsonString, int tenantId) throws APIException {
+
+        String resourcePath = SwaggerConstants.DEFAULT_SWAGGER_REGISTRY_PATH + apiName + "/swagger.json";
+
+        RegistryService registryService = RegistryServiceHolder.getInstance().getRegistryService();
+        try {
+            Registry registry = registryService.getConfigSystemRegistry(tenantId);
+
+            org.wso2.carbon.registry.core.Resource resource;
+            if (!registry.resourceExists(resourcePath)) {
+                resource = registry.newResource();
+                resource.setContent(swaggerJsonString);
+                resource.setMediaType("application/json");
+            } else {
+                resource = registry.get(resourcePath);
+                resource.setContent(swaggerJsonString);
+            }
+            registry.put(resourcePath, resource);
+
+        } catch (RegistryException e) {
+            handleException(log, "Could not update swagger document", e);
+        }
+    }
+
+    /**
+     * Return the registry resource for the provided location
+     *
+     * @param apiName
+     * @param tenantId
+     * @return
+     * @throws APIException
+     */
+    public String getSwaggerDocument(String apiName, int tenantId) throws APIException {
+
+        API api = getSynapseAPIByName(apiName);
+        if (api == null) {
+            throw new APIException("API with name \"" + apiName + "\" does not exists.");
+        }
+
+        String resourcePath = SwaggerConstants.DEFAULT_SWAGGER_REGISTRY_PATH + apiName + "/swagger.json";
+        String swaggerJsonString = null;
+        try {
+            swaggerJsonString = getResourceFromRegistry(resourcePath, tenantId);
+        } catch (RegistryException e) {
+            handleException(log, "Could not get swagger document", e);
+        }
+
+        // Generate if not available
+        if (swaggerJsonString == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Generate swagger definition for the API : " + apiName);
+            }
+
+            try {
+                JSONObject jsonDefinition = new JSONObject(new GenericApiObjectDefinition(api).getDefinitionMap());
+                swaggerJsonString = jsonDefinition.toString();
+            } catch (AxisFault axisFault) {
+                handleException(log, "Error occurred while generating swagger definition", axisFault);
+            }
+        }
+        return swaggerJsonString;
+    }
+
+    /**
+     * Function to generate API from swagger definition (from JSON representation)
+     *
+     * @param swaggerJsonString swagger definition
+     * @return generated synapse API
+     * @throws APIException
+     */
+    public String generateAPIFromSwagger(String swaggerJsonString) throws APIException {
+
+        if (swaggerJsonString == null || swaggerJsonString.isEmpty()) {

```

Comment:
AFAIR we can use StringUtils.isEmpty()

Reply:
We have to depend on a dependency just for this check. Hence shall we avoid adding a dependency for apache commons-lang

Reply:
+1